pkg/vsock/handshake_windows.go

```diff
@@ -48,76 +49,80 @@ func GetVMGUID(seed string, handshakePort uint32, timeout <-chan time.Time) (hvs
 
 	key.Close()
 
+	ctx, cancel := context.WithCancel(ctx)
 	found := make(chan hvsock.GUID, len(names))
-	done := make(chan bool, len(names))
-	defer close(done)
 
 	for _, name := range names {
 		vmGUID, err := hvsock.GUIDFromString(name)
 		if err != nil {
 			logrus.Errorf("invalid VM name: [%s], err: %v", name, err)
 			continue
 		}
-		go handshake(vmGUID, seed, handshakePort, found, done)
+		go handshake(ctx, vmGUID, handshakePort, signature, found)
 	}
-	return tryFindGUID(found, timeout)
+	return tryFindGUID(cancel, found, timeout)
 }
 
-// handshake attempts to perform a handshake by verifying the seed with a running
-// af_vsock peer in WSL distro, it attempts once per second
-func handshake(vmGUID hvsock.GUID, seedPhrase string, peerHandshakePort uint32, found chan<- hvsock.GUID, done <-chan bool) {
-	svcPort, err := hvsock.GUIDFromString(winio.VsockServiceID(peerHandshakePort).String())
+// GetVsockConnection estabilishes a new AF_VSOCK connection with
+// the provided VM GUID and port, the caller is responsible for closing the connection
+func GetVsockConnection(vmGUID hvsock.GUID, port uint32) (net.Conn, error) {
+	svcPort, err := hvsock.GUIDFromString(winio.VsockServiceID(port).String())
 	if err != nil {
-		logrus.Errorf("hostHandshake parsing svc port: %v", err)
+		return nil, err
 	}
 	addr := hvsock.Addr{
 		VMID:      vmGUID,
 		ServiceID: svcPort,
 	}
 
+	return hvsock.Dial(addr)
+}
+
+// handshake with the Hyper-V VM by verifying the fixed signature over AF_VSOCK once
+// per second, in order to identify the VM running the WSL distro.
+func handshake(ctx context.Context, vmGUID hvsock.GUID, peerHandshakePort uint32, signaturePhrase string, found chan<- hvsock.GUID) {
 	attempInterval := time.NewTicker(time.Second * 1)
-	attempt := 1
+	attempt := 0
 	for {
 		select {
-		case <-done:
+		case <-ctx.Done():
 			logrus.Infof("attempt to handshake with [%s], goroutine is terminated", vmGUID.String())
 			return
 		case <-attempInterval.C:
-			// Spawn a goroutine here to ensure we don't get stuck on a timeout
 			go func() {
-				conn, err := hvsock.Dial(addr)
+				conn, err := GetVsockConnection(vmGUID, peerHandshakePort)
 				select {
-				case <-done:
-					// If we're already connected, no need to print more things.
+				case <-ctx.Done():
 					return
 				default:
 				}
 				if err != nil {
 					attempt++
-					logrus.Debugf("handshake attempt[%v] to dial into VM [%s], looking for vsock-peer", attempt, vmGUID.String())
+					logrus.Errorf("handshake attempt[%v] to dial into VM [%s], looking for vsock-peer failed: %v", attempt, vmGUID.String(), err)
```

Comment:
Is this an _error_ level event? This means that we'll have a bunch of logs even without `-debug` that are not helpful…